components/data-feed-service/service/data-feed-aggregate-task_test.go

```diff
@@ -0,0 +1,41 @@
+package service
+
+import (
+	"context"
+	"testing"
+)
+
+const (
+	ipAddress = "192.168.10.10"
+	nodeId    = "e5a3d952-5dfc-3215-bfe4-541eb0b5d345"
+)
+
+func TestIsIpaddressTrue(t *testing.T) {
+	isIPAddress := isIPAddress(ipAddress)
+	if !isIPAddress {
+		t.Log("Expected true got false")
+		t.Fail()
+	}
+}
+
+func TestIsIpaddressFalse(t *testing.T) {
+	isIPAddress := isIPAddress(nodeId)
+	if isIPAddress {
+		t.Log("Expected false got true")
+		t.Fail()
+	}
+}
+
+func TestSafeGetNodeDataNoIPFilter(t *testing.T) {
+	aggregateTask := NewDataFeedAggregateTask(mockConfig, nil, nil, nil, nil)
+	var filters []string
+	nodeData, err := aggregateTask.safeGetNodeData(context.Background(), filters)
+	if nodeData == nil || len(nodeData) != 0 {
+		t.Logf("Expected empty nodeData, got %v", nodeData)
+		t.Fail()
+	}
+	if err != nil {
+		t.Logf("Expected nil, got %v", err)
+		t.Fail()
+	}
```

Comment:
With the package mentioned above (its `require` variant), we could make this say
```suggestion
	nodeData, err := aggregateTask.safeGetNodeData(context.Background(), filters)
	require.NoError(t, err)
	require.NotNil(t, nodeData)
	require.NotEmpty(t, nodeData)
```

Reply:
Nice, I'll start using that next time I add some more tests.